internal/pkg/fieldtags/analyzer.go

```diff
@@ -0,0 +1,114 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package fieldtags defines an analyzer that identifies struct fields tagged with datapol annotations
+// indicating that they contain SPII.
+package fieldtags
+
+import (
+	"go/ast"
+	"reflect"
+	"strings"
+
+	"golang.org/x/tools/go/analysis"
+	"golang.org/x/tools/go/analysis/passes/inspect"
+	"golang.org/x/tools/go/analysis/passes/structtag"
+	"golang.org/x/tools/go/ast/inspector"
+)
+
+type keyValue struct {
+	key   string
+	value string
+}
+
+type sourcePatterns []keyValue
+
+var patterns sourcePatterns = []keyValue{
+	{"levee", "source"},
+}
+
+var Analyzer = &analysis.Analyzer{
+	Name: "fieldtags",
+	Doc:  "This analyzer identifies Source fields based on their tags.",
+	Run:  run,
+	Requires: []*analysis.Analyzer{
+		inspect.Analyzer,
+		structtag.Analyzer, // required to ensure that tags are properly formatted
```

Comment:
This doesn't appear to do what we thought.  Case in point, I was wrong about the tag convention in my other comment, and
```
  var int "levee:`source`"
```
is not properly tagged.

```bash
$ .../go-flow-levee > go vet -structtag internal/pkg/fieldtags/testdata/src/tests/test.go
# command-line-arguments
internal/pkg/fieldtags/testdata/src/tests/test.go:19:2: struct field secret has json tag but is not exported
internal/pkg/fieldtags/testdata/src/tests/test.go:20:2: struct field tag "levee:`source`" not compatible with reflect.StructTag.Get: bad syntax for struct tag value
```

Yet this doesn't appear to emit the finding as an error nor does it expose a `Result`.  I'm not sure there's a way to get a grip on that analyzer's output...

Given that, I think it could be alright if we say that tags must satisfy the `go vet -structtag` format, but not explicitly check for it since we can't get our hooks into it.

Reply:
That's kind of odd. I think it's reasonable to expect that the field tags will be properly formatted. Too bad we can't verify it in advance, though.

I've added a sentence to the doc to reflect our expectation.